### ISSUE TITLE
Tracking issue for r10

### ISSUE THREAD

**Issue by 9ary** [redolution/gekkoboot#40]
- ~~Split iplboot into stages 1 (modchip flash, bare minimum to load a DOL from local storage) and 2 (config support, USB/network boot, controller shortcuts...)~~
- [ ] Implement preloader stage (no libc/libogc2, can boot a recovery tool from SD card or chainload to gekkoboot proper)
- [ ] Only disable Qoob in Qoob builds, check what needs to be done before that, and whether it's necessary for Qoob SX
- [ ] Implement our own, source-built BS1 replacement
- ~~Deferred video init - initialize video at the last moment~~ there isn't really a reasonable way to do this
- [x] No media/no stage 2 DOL should be a hard error and halt, rather than reset to the IPL
- [ ] Configuration file using [INIH](https://github.com/benhoyt/inih) or https://github.com/compuphase/minIni
  - Support glob patterns using fnmatch
  - Remove all the hardcoded paths
  - Shortcut timeout
  - Configurable remote boot priority and timeout
- [ ] Stealth mode - SRAM setting (`SYS_Get/SetBootMode`) to skip iplboot by default
- [ ] Use FatFs more correctly, replace the current IFACE global with FF_STR_VOLUME_ID
- [ ] Fix #25
- [ ] Look into #26, it should at least not crash
- [ ] Improve logging
- [ ] #44
